tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -56,33 +57,38 @@ class DistributedTransformerTest
 };
 
 namespace {
+// testValidate doesn't work out of the box due to #2906, so I had to manually
+// specify the tolerances.
 void validate(
-    std::vector<at::Tensor> expected_out,
-    std::vector<at::Tensor> out) {
-  EXPECT_EQ(expected_out.size(), out.size());
-  for (auto i : c10::irange(out.size())) {
+    const std::vector<at::Tensor>& expected_outputs,
+    const std::vector<at::Tensor>& outputs,
+    const std::vector<double>& atols) {
+  using testing::SizeIs;
+  const auto num_outputs = outputs.size();
+  ASSERT_THAT(expected_outputs, SizeIs(num_outputs));
+  ASSERT_THAT(atols, SizeIs(num_outputs));
+
+  for (const auto i : c10::irange(num_outputs)) {
     // allclose can catch this as well. However, it would throw an exception,
     // not showing which output was problematic.
-    ASSERT_EQ(out[i].dtype(), expected_out[i].dtype())
+    ASSERT_EQ(outputs[i].dtype(), expected_outputs[i].dtype())
         << "Output " << i << " has a mismatching data type.";
 
     // Note: Scaling tolerance up since the error accumulates across ops
     // BFloat16 error is quite high, but the program has been verified with
     // double precision to be logically correct.
-    const double atol = 0.075 * (i + 1);
-    const double rtol = 1.6e-2;
```

Comment:
It would also be beneficial to keep some form of default value, since tests are still getting added this adds to a lot of effort to get a simple example working.

Reply:
I removed it because I found I ended up always setting rtol to 0 :) 

I'd love to keep some default value to, as you said, minimize the effort to get a simple example working. What would be a good default value? The old `atol=0.075*(i+1)` is too relaxed after I fixed the reference implementation. The `*(i+1)` part is also problematic because the output number doesn't necessarily match the layer number. 

How about this: 
1. Hardcode rtol to 0.016. I'll try to finetune this value a little bit -- 0.016 sounds like a large **default** rtol to start with. 
2. Still require the caller to provide a list of per-output atols, because it's hard to fine a good default and each test seems to require something different.

Reply:
It was supposed to have been taken from Pytorch. https://github.com/pytorch/pytorch/blob/042f2f7746a064f1527d95d1f1d712b4f0b34186/test/test_transformers.py#L85
But should be .0126.

Reply:
Done -- added default rtols.